nixos/modules/security/duosec.nix

```diff
@@ -195,21 +184,49 @@ in
   };
 
   config = mkIf (cfg.ssh.enable || cfg.pam.enable) {
-     environment.systemPackages = [ pkgs.duo-unix ];
-
-     security.wrappers.login_duo.source = "${pkgs.duo-unix.out}/bin/login_duo";
-     environment.etc = loginCfgFile // pamCfgFile;
-
-     /* If PAM *and* SSH are enabled, then don't do anything special.
-     If PAM isn't used, set the default SSH-only options. */
-     services.openssh.extraConfig = mkIf (cfg.ssh.enable || cfg.pam.enable) (
-     if cfg.pam.enable then "UseDNS no" else ''
-       # Duo Security configuration
-       ForceCommand ${config.security.wrapperDir}/login_duo
-       PermitTunnel no
-       ${optionalString (!cfg.allowTcpForwarding) ''
-         AllowTcpForwarding no
-       ''}
-     '');
+    environment.systemPackages = [ pkgs.duo-unix ];
+
+    security.wrappers.login_duo.source = "${pkgs.duo-unix.out}/bin/login_duo";
+
+    system.activationScripts = {
+      login_duo = mkIf cfg.ssh.enable ''
+        mkdir -m 0755 -p /etc/duo
+
+        umask 277
```

Comment:
I'm not sure how isolated activation scripts are.  Something to look in to and maybe not necessary, but my first thought was "oh, mayb ethat should be wrapped in a subshell with `(...this code ...)` to prevent umask changes spreading." But now that I type it out, I feel pretty sure the isolation is fine... :)

Reply:
Specifically for this PR if a sysadmin has `ssh.enable = true; pam.enable = false;` and deploys, followed by `ssh.enable = false; pam.enable = true;` this will result in files lingering under `/etc/duo` and produce the unintended result.

As a bare minimum the script should be cleaning up files which aren't meant to be there on every deploy. I just want to know to what degree I should go to, and what the expectation is.

Thanks for feedback @grahamc!